26-dracula-ant/technical_specification.md

```diff
@@ -99,31 +122,38 @@ The database of the Work Package Service should store the following WorkPackage
 ```python
 WorkPackage:
   id: str  # the ID of this work package
-  user_id: str  # the internal user ID
   dataset_id: str  # all files must belong to the dataset with this ID
   type: enum  # the work order type, either download or upload
-  file_ids: list[str]  # a list of IDs for all files that shall be downloaded or uploaded
-  public_key: str  # the user's public Crypt4GH key
+  files: dict[str, str]  # IDs of all included files mapped to their extensions
+  user_id: str  # the internal user ID
   full_user_name: str  # the full name of the user (with academic title)
-  email: str  # he email address of the user
+  email: str  # the email address of the user
+  user_public_crypt4gh_key: str  # the user's public Crypt4GH key
   token_hash: str  # hash of the work package access token
   created: datetime  # creation date of this work package
   expires: datetime  # expiry date of this work package
 ```
 
-The `file_ids` list must not be empty. It contains either the `file_ids` specified by the user when creating the work package (in this case the Work Package Service checks whether they belong to the given `dataset_id`), or the full list of all `file_id`s that belong to the given `dataset_id`.
+The `files` mapping must not be empty. It contains either the `file_ids` specified by the user when creating the work package (in this case the Work Package Service checks whether they belong to the given `dataset_id`), or the full list of all `file_id`s that belong to the given `dataset_id` if the user didn't specify a list of file IDs. The values of this mapping are the file extensions including the dot.
 
 An entry in the `WorkPackage` collection is only created by the Work Package Service after verification that the user with the given `user_id` is allowed to access the dataset with the given `dataset_id` (see access checks above).
 
-In the database, the Work Package Service also keeps track of which file IDs belong to which dataset and which users are allowed to submit data for the individual datasets. The following association collection is used for this purpose:
+In the database, the Work Package Service also keeps track of datasets, particularly which file IDs belong to which dataset, how the datasets and files are named, (and maybe also which users are allowed to submit data for the individual datasets). The following association collection is used for this purpose:
```

Comment:
Filenames are not shared with the requester. I would not duplicate permission information. The claims repo and tge visa issuer should be responsible for that.

Reply:
Ok, formulated that it a bit more clearly.

I'm not sure if access permission to *submit* data should also be part of the claims repo. Do you think so? (Then at least I would use a custom visa type for that.)

Reply:
That might be a good implementation. However, we discuss that as part of a different epic.